src/coord/cartesian/Cartesian2D.js

```diff
@@ -81,7 +81,15 @@ Cartesian2D.prototype = {
      * @return {boolean}
      */
     containData: function (data) {
-        return this.getAxis('x').containData(data[0])
+        var axisX = this.getAxis('x');
+
+        // For aixs type is category,
+        // only one data shows on viewport just check if it xAxis' value equal to scale
+        if (axisX.type === 'category' && axisX.scale.count() === 1) {
+            return data[0] === axisX.scale._extent[0];
```

Comment:
`_extent` is a private variable. You should probably call `axisX.scale.getExtent()[0]` to get it.